### PR TITLE
Recommend the Visual Studio SPA project templates

### LINKED ISSUE #30155: Update ASP.NET Core SPA content for .NET 8
In .NET 8 we've removed the ASP.NET Core SPA templates from the .NET SDK in favor of the newer Visual Studio SPA templates. We need to update the ASP.NET Core SPA docs for .NET 8 docs accordingly, linking to the Visual Studio SPA docs as appropriate:

- [JavaScript and TypeScript in Visual Studio overview](https://learn.microsoft.com/visualstudio/javascript/javascript-in-visual-studio)
- [Tutorial: Create an ASP.NET Core app with React in Visual Studio](https://learn.microsoft.com/visualstudio/javascript/tutorial-asp-net-core-with-react)
- [Tutorial: Create an ASP.NET Core app with Angular in Visual Studio](https://learn.microsoft.com/visualstudio/javascript/tutorial-asp-net-core-with-angular)
- [Tutorial: Create an ASP.NET Core app with Vue in Visual Studio](https://learn.microsoft.com/visualstudio/javascript/tutorial-asp-net-core-with-vue)
- [Tutorial: Create an ASP.NET Core app with TypeScript in Visual Studio](https://learn.microsoft.com/visualstudio/javascript/tutorial-aspnet-with-typescript)

@joj @jiayac

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: c1ffb8e1-215c-c148-0b28-da0d62665fa4
* Version Independent ID: c1ffb8e1-215c-c148-0b28-da0d62665fa4
* Content: [Overview of Single Page Applications (SPA) in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/client-side/spa/intro?view=aspnetcore-7.0)
* Content Source: [aspnetcore/client-side/spa/intro.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/client-side/spa/intro.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-clientside**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **jacalvar**

---
[Associated WorkItem - 149989](https://dev.azure.com/msft-skilling/Content/_workitems/edit/149989)

### LINKED ISSUE #30160: .NET 8: Update React to use new templates
[Project Templates](https://devblogs.microsoft.com/visualstudio/exploring-javascript-and-typescript-development-in-visual-studio/#project-templates)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 261c9875-3db6-4d30-f9d6-922a9334ecb2
* Version Independent ID: b8f8f7c4-9c90-869d-9fd7-a887bc28f262
* Content: [Use React with ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/client-side/spa/react?view=aspnetcore-7.0&tabs=visual-studio)
* Content Source: [aspnetcore/client-side/spa/react.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/client-side/spa/react.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-clientside**
* GitHub Login: @SteveSandersonMS
* Microsoft Alias: **scaddie**

---
[Associated WorkItem - 159221](https://dev.azure.com/msft-skilling/Content/_workitems/edit/159221)

### LINKED ISSUE #30274: VS template for Angular
[Project Templates](https://devblogs.microsoft.com/visualstudio/exploring-javascript-and-typescript-development-in-visual-studio/#project-templates)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 32a18270-91cd-2a02-2eb8-2c8b1ed03737
* Version Independent ID: c58d57e5-a0c2-201d-f6dd-076ad5971fcb
* Content: [Use Angular with ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/client-side/spa/angular?view=aspnetcore-7.0&tabs=visual-studio)
* Content Source: [aspnetcore/client-side/spa/angular.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/client-side/spa/angular.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-clientside**
* GitHub Login: @SteveSandersonMS
* Microsoft Alias: **stevesa**

---
[Associated WorkItem - 159220](https://dev.azure.com/msft-skilling/Content/_workitems/edit/159220)

### LINKED ISSUE #30275: Create an ASP.NET Core app with Vue
[Project Templates](https://devblogs.microsoft.com/visualstudio/exploring-javascript-and-typescript-development-in-visual-studio/#project-templates)

image

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 32a18270-91cd-2a02-2eb8-2c8b1ed03737
* Version Independent ID: c58d57e5-a0c2-201d-f6dd-076ad5971fcb
* Content: [Use Angular with ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/client-side/spa/angular?view=aspnetcore-7.0&tabs=visual-studio)
* Content Source: [aspnetcore/client-side/spa/angular.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/client-side/spa/angular.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-clientside**
* GitHub Login: @SteveSandersonMS
* Microsoft Alias: **stevesa**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #30275 
Fixes #30274 
Fixes #30155 
Fixes #30160

For [Authentication and authorization for SPAs](https://learn.microsoft.com/aspnet/core/security/authentication/identity-api-authorization), this PR only fixes a moniker prep error; the content of the article will be updated for #29363 and #30066

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/client-side/spa-services.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/client-side/spa-services.md) | [Use JavaScript Services to Create Single Page Applications in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/client-side/spa-services?branch=pr-en-us-30333) |
| [aspnetcore/client-side/spa/angular.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/client-side/spa/angular.md) | [[Visual Studio](#tab/visual-studio/)](https://review.learn.microsoft.com/en-us/aspnet/core/client-side/spa/angular?branch=pr-en-us-30333) |
| [aspnetcore/client-side/spa/intro.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/client-side/spa/intro.md) | [Overview of Single Page Applications (SPA) in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/client-side/spa/intro?branch=pr-en-us-30333) |
| [aspnetcore/client-side/spa/react.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/client-side/spa/react.md) | [Use React with ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/client-side/spa/react?branch=pr-en-us-30333) |
| [aspnetcore/fundamentals/middleware/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/fundamentals/middleware/index.md) | [aspnetcore/fundamentals/middleware/index](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/middleware/index?branch=pr-en-us-30333) |
| [aspnetcore/introduction-to-aspnet-core.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/introduction-to-aspnet-core.md) | [Overview of ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/introduction-to-aspnet-core?branch=pr-en-us-30333) |
| [aspnetcore/security/authentication/identity-api-authorization.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/security/authentication/identity-api-authorization.md) | [Authentication and authorization for SPAs](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?branch=pr-en-us-30333) |
| [aspnetcore/tutorials/choose-web-ui.md](https://github.com/dotnet/AspNetCore.Docs/blob/b2e9aa9854704e199b257ebe95f27f737bd7ae86/aspnetcore/tutorials/choose-web-ui.md) | [Choose an ASP.NET Core UI](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/choose-web-ui?branch=pr-en-us-30333) |

### CODE REVIEW

File: aspnetcore/client-side/spa/angular.md
Comment (Rick-Anderson) [inline aspnetcore/client-side/spa/angular.md:21]: I think we're better off to have to TOC go directly the VS content. Why keep the old content and not just go directly to VS? I'll leave it to @danroth27

**Review by Rick-Anderson — COMMENTED** [review]
I think we should just go to VS on all of these
image

File: aspnetcore/client-side/spa/angular.md
Comment (tdykstra) [reply aspnetcore/client-side/spa/angular.md:21]: Some people will still want to use the old templates, and this way they can get the documentation for them by selecting the 7.0 version of the docs. And they find out about that availability when they are in the 8.0 version of the docs. We have TOC entries that go directly to the Visual Studio tutorials, and they're listed first, and their link text identifies them as VS tutorials.

Comment (tdykstra) [reply aspnetcore/client-side/spa/angular.md:21]: @Rick-Anderson One way to address your concern is to set the file-level moniker for the SPA Overview, Angular, and React articles to < 9.0. Then they will automatically drop out of the TOC anytime someone selects a 9.0 moniker. That gives people time to switch to the new templates without losing TOC access to docs for the old ones, but lets us get the old ones out of the TOC when .NET 9 is selected, without us having to worry about creating orphaned articles by just deleting them from the TOC. We already have the TOC links to the VS tutorials, and they will stay there.

**Comment by tdykstra** [comment]
@danroth27 Please review

File: aspnetcore/includes/spa-templates-recommend-vs.md
Comment (danroth27) [inline aspnetcore/includes/spa-templates-recommend-vs.md:11]: There are template options for using both an API controller or a minimal API.

```suggestion
* Use an ASP.NET Core project for the backend.
```

File: aspnetcore/includes/spa-templates-new-vs-old.md
Comment (danroth27) [inline aspnetcore/includes/spa-templates-new-vs-old.md:9]: The new SPA templates that include an ASP.NET Core backend will ship their first stable release with 17.8. Also, let's not draw unnecessary attention to the lack of CLI support. We think the existing SPA templates saw very little usage from the CLI. If it turns out users want CLI support for project creation, we'll consider adding it.

 ```suggestion
The SPA templates that use the special project type for JavaScript and TypeScript are available in Visual Studio 2022 version 17.8 or later with the **ASP.NET and web development** workload installed.
```

File: aspnetcore/includes/spa-templates-new-vs-old.md
Comment (danroth27) [inline aspnetcore/includes/spa-templates-new-vs-old.md:11]: Can we reword this part to not be a comparison? In the long run the intention is that the new templates fully replace the old ones. Can we make the language here timeless?

```suggestion
The Visual Studio templates for building ASP.NET Core apps with a JavaScript or TypeScript frontend offer the following benefits:
```

File: aspnetcore/includes/spa-templates-new-vs-old.md
Comment (danroth27) [inline aspnetcore/includes/spa-templates-new-vs-old.md:7]:
```suggestion
## ASP.NET Core SPA templates
```

File: aspnetcore/includes/spa-templates-new-vs-old.md
Comment (danroth27) [inline aspnetcore/includes/spa-templates-new-vs-old.md:25]:
```suggestion
## Legacy ASP.NET Core SPA templates

Earlier versions of the .NET SDK included what are now legacy templates for building SPA apps with ASP.NET Core. For documentation on these older templates, see the ASP.NET Core 7.0 version of the [SPA overview](xref:spa/angular?view=aspnetcore-7.0&preserve-view=true&tabs=netcore-cli) and the [Angular](xref:spa/angular?view=aspnetcore-7.0&preserve-view=true&tabs=netcore-cli) and [React](xref:spa/react?view=aspnetcore-7.0&preserve-view=true&tabs=netcore-cli) articles.
```

File: aspnetcore/client-side/spa/angular.md
Comment (danroth27) [reply aspnetcore/client-side/spa/angular.md:21]: I wanted to have a discoverable way in the ASP.NET Core doc set for building SPA based apps. I'm assuming that many ASP.NET Core devs will start in the ASP.NET Core docs and will look there for SPA support. So, as long as we have a way to discover the SPA content from the ASP.NET Core docs and a way for existing users to find the old content, I'm happy.

File: aspnetcore/includes/spa-templates-recommend-vs.md
Comment (danroth27) [inline aspnetcore/includes/spa-templates-recommend-vs.md:7]:
```suggestion
Visual Studio provides project templates for creating single-page apps (SPAs) based on JavaScript frameworks such as [Angular](https://angular.io/), [React](https://facebook.github.io/react/), and [Vue](https://vuejs.org/) that have an ASP.NET Core backend. These templates:
```

File: aspnetcore/fundamentals/middleware/index.md
Comment (danroth27) [inline aspnetcore/fundamentals/middleware/index.md:204]: Why not just link to the SPA intro here?

File: aspnetcore/introduction-to-aspnet-core.md
Comment (danroth27) [inline aspnetcore/introduction-to-aspnet-core.md:45]: I'd like to elevate Blazor here as our built-in full stack offering.

```suggestion
ASP.NET Core includes [Blazor](xref:blazor/index) for building richly interactive web UI, and also integrates with other popular frontend JavaScript frameworks like [Angular](/visualstudio/javascript/tutorial-asp-net-core-with-angular), [React](/visualstudio/javascript/tutorial-asp-net-core-with-react), [Vue](/visualstudio/javascript/tutorial-asp-net-core-with-vue), and [Bootstrap](https://getbootstrap.com/). For more information, see <xref:blazor/index> and related topics under *Client-side development*.
```

File: aspnetcore/introduction-to-aspnet-core.md
Comment (danroth27) [inline aspnetcore/introduction-to-aspnet-core.md:124]: Isn't this version of the text specific to .NET Core versions prior to 3.0? Blazor didn't ship until .NET Core 3.0.

**Review by danroth27 — APPROVED** [review]
Some minor nits, but otherwise looks good.

File: aspnetcore/fundamentals/middleware/index.md
Comment (tdykstra) [reply aspnetcore/fundamentals/middleware/index.md:204]:
```suggestion
For information about Single Page Applications, see <xref:spa/intro>.
```

File: aspnetcore/introduction-to-aspnet-core.md
Comment (tdykstra) [reply aspnetcore/introduction-to-aspnet-core.md:124]:
```suggestion
ASP.NET Core integrates seamlessly with popular client-side frameworks and libraries, including [Angular](/visualstudio/javascript/tutorial-asp-net-core-with-angular), [React](/visualstudio/javascript/tutorial-asp-net-core-with-react), and [Bootstrap](https://getbootstrap.com/). For more information, see the topics under *Client-side development*.
```

**Pull request merged**
